onebusaway-enterprise-webapp/src/main/java/org/onebusaway/enterprise/webapp/actions/api/StopsOnRouteAction.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (C) 2011 Metropolitan Transportation Authority
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.onebusaway.enterprise.webapp.actions.api;
+
+import org.apache.struts2.convention.annotation.ParentPackage;
+import org.apache.struts2.convention.annotation.Result;
+import org.onebusaway.enterprise.webapp.actions.OneBusAwayEnterpriseActionSupport;
+import org.onebusaway.enterprise.webapp.actions.api.model.StopOnRoute;
+import org.onebusaway.transit_data.model.*;
+import org.onebusaway.transit_data.services.TransitDataService;
+import org.onebusaway.transit_data_federation.services.AgencyAndIdLibrary;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@ParentPackage("json-default")
+@Result(type="json", params={"callbackParameter", "callback"})
+public class StopsOnRouteAction extends OneBusAwayEnterpriseActionSupport {
+
+  private static final long serialVersionUID = 1L;
+
+  @Autowired
+  private TransitDataService _transitDataService;
+
+  private List<StopOnRoute> _stops = new ArrayList<StopOnRoute>();
```

Comment:
I think this wants to be a Set, not a List, as you may be iterating over multiple stop groupings and therefor may have duplicate stops